packages/okta-react/README.md

```diff
@@ -97,69 +128,106 @@ class App extends Component {
 export default App;
 ```
 
-## Show Login and Logout Buttons
+#### Creating React Router Routes with function-based components
 
-In the relevant location in your application, you will want to provide `Login` and `Logout` buttons for the user. You can show/hide the correct button by using the `auth.isAuthenticated()` method. For example:
+```jsx
+import React from 'react';
+import { SecureRoute } from '@okta/okta-react/react-router';
+import { Security, LoginCallback } from '@okta/okta-react';
+import Home from './Home';
+import Protected from './Protected';
 
+const App = () => (
+  <Router>
+    <RouterSecurity
+      issuer='https://{yourOktaDomain}.com/oauth2/default'
+      clientId='{clientId}'
+      redirectUri={window.location.origin + '/implicit/callback'}
+    >
+      <Route path='/' exact={true} component={Home}/>
+      <SecureRoute path='/protected' component={Protected}/>
+      <Route path='/implicit/callback' component={ImplicitCallback} />
+    </RouterSecurity>
+  </Router>
+);
+
+export default App;
+```
+
+#### Show Login and Logout Buttons (class-based)
 ```jsx
 // src/Home.js
 
 import React, { Component } from 'react';
-import { withAuth } from '@okta/okta-react';
+import { withOktaAuth } from '@okta/okta-react';
 
-export default withAuth(class Home extends Component {
+export default withOktaAuth(class Home extends Component {
   constructor(props) {
     super(props);
-    this.state = { authenticated: null };
-    this.checkAuthentication = this.checkAuthentication.bind(this);
     this.login = this.login.bind(this);
     this.logout = this.logout.bind(this);
   }
 
-  async checkAuthentication() {
-    const authenticated = await this.props.auth.isAuthenticated();
-    if (authenticated !== this.state.authenticated) {
-      this.setState({ authenticated });
-    }
-  }
-
   async login() {
-    this.props.auth.login('/');
+    this.props.authService.login('/');
   }
 
   async logout() {
-    this.props.auth.logout('/');
-  }
-
-  async componentDidMount() {
-    this.checkAuthentication();
-  }
-
-  async componentDidUpdate() {
-    this.checkAuthentication();
+    this.props.authService.logout('/');
   }
 
   render() {
-    if (this.state.authenticated === null) return null;
-    return this.state.authenticated ?
+    if (this.props.authState.isPending) return <div>Loading...</div>;
```

Comment:
This whole example is MUCH improved by these authState changes! I hated that `checkAuthentication` on `componentDidUpdate`! Our users are going to love this.